core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/AwsXmlProtocolFactory.java

```diff
@@ -163,10 +163,15 @@ Optional<XmlElement> getErrorRoot(XmlElement document) {
 
     private XmlGenerator createGenerator(OperationInfo operationInfo) {
         return operationInfo.hasPayloadMembers() ?
-               XmlGenerator.create(operationInfo.addtionalMetadata(XML_NAMESPACE_ATTRIBUTE)) :
+               XmlGenerator.create(operationInfo.addtionalMetadata(XML_NAMESPACE_ATTRIBUTE), isS3(operationInfo)) :
                null;
     }
 
+    private boolean isS3(OperationInfo operationInfo) {
+        return "http://s3.amazonaws.com/doc/2006-03-01/"
```

Comment:
Is that the only way we can check for s3? Can the namespace change? What happens if it changes?

Reply:
namespace seems to be the only way as far as I can tell
good point, updated to check for the prefix without the date and extracted to a class field for more visibility